cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -23,13 +23,14 @@ import (
 type MasterConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
-	API               API        `json:"api"`
-	Discovery         Discovery  `json:"discovery"`
-	Etcd              Etcd       `json:"etcd"`
-	Networking        Networking `json:"networking"`
-	KubernetesVersion string     `json:"kubernetesVersion"`
-	CloudProvider     string     `json:"cloudProvider"`
-	AuthorizationMode string     `json:"authorizationMode"`
+	API                API               `json:"api"`
+	Discovery          Discovery         `json:"discovery"`
+	Etcd               Etcd              `json:"etcd"`
+	Networking         Networking        `json:"networking"`
+	KubernetesVersion  string            `json:"kubernetesVersion"`
+	CloudProvider      string            `json:"cloudProvider"`
+	AuthorizationMode  string            `json:"authorizationMode"`
+	ExtraAPIServerArgs map[string]string `json:"extraAPIServerArgs"`
```

Comment:
I would name this "APIServerExtraFlags"

Put API server first so that as we have other stuff it sorts.

Flags explains why it is a map. Still not clear how to do singleton flags or remove flags but I think that is okay.

Also, we should start documenting these structures (ala types.go everywhere else).  Explain that this is a way to add and override flags that get set on the API server.

For things like the admission controller this still isn't 100% as there is no way to add/remove/modify the list.  But it is good enough to get started for now and at least it is possible to modify this stuff.

Might also be good to support this for the KCM and the scheduler.  But that can come in other PRs.